joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -595,7 +599,7 @@ CallbackReturn JointTrajectoryController::on_configure(const rclcpp_lifecycle::S
 
   if (has_velocity_state_interface_)
   {
-    if (!contains_interface_type(state_interface_types_, hardware_interface::HW_IF_POSITION))
+    if (!has_position_state_interface_)
```

Comment:
Investigate how to refactor these cascading `if`s so that it is easier to read which conditions together lead to a failure.
This section of  the code looks like a table of booleans where in some cases an error is flagged up.